### PR TITLE
When generate doc, ignore the hidden parameters.

### LINKED ISSUE #15771: [Doc] Auto-hide deprecated parameters in Pulsar admin CLI docs
### What issue do you find in Pulsar docs?

In Pulsar admin CLI docs, some commands have duplicate parameters. For example, processing guarantees have duplicate parameters as shown below:
image

However, in the code file, we can tell that `--processingGuarantees` has been deprecated and this information is not demonstrated in docs which may confuse users.
image

### What is your suggestion?

Auto-hide deprecated parameters in Pulsar admin CLI docs by enhancing the auto-generation mechanism.

### Do you have any references?

https://pulsar.apache.org/tools/pulsar-admin/2.10.0-SNAPSHOT/#-em-localrun-em-
https://github.com/apache/pulsar/blob/branch-2.10/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java#L279-L280

### Would you like to fix this issue?

Ping @urfreespace @tuteng 
//cc @Anonymitaet 

### Note

- [X] I have researched my question.

### PR DESCRIPTION
**Pull request by horizonzy**
Fixes #15771

### Documentation

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `doc-not-needed` 
(Please explain why)
  
- [x] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Comment by urfreespace** [comment]
> @urfreespace pls check it, thanks.

pulsar-admin/pulsar-client/pulsar-perf/pulsar all need to be fixed, I think they are different cmd class file

**Comment by horizonzy** [comment]
Run pulsar-admin configFile documents generate functions at local.
Before this change output:
[Before.txt](https://github.com/apache/pulsar/files/9017600/Before.txt)

After this change output:
[After.txt](https://github.com/apache/pulsar/files/9017599/After.txt)

**Comment by horizonzy** [comment]
> > @urfreespace pls check it, thanks.
> 
> pulsar-admin/pulsar-client/pulsar-perf/pulsar all need to be fixed, I think they are different cmd class file

Thanks your reminder, the new commit pushed.

**Comment by horizonzy** [comment]
@Anonymitaet @momo-jun Pls confirm shall we ignore the `ServiceConfiguration` deprecated param.

**Review by urfreespace — APPROVED** [review]

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/BaseGenerateDocumentation.java [hunk 1]
```diff
@@ -90,7 +90,7 @@ protected String generateDocByFieldContext(String className, String type, String
         sb.append("|---|---|---|---|---|\n");
         for (Field field : fields) {
             FieldContext fieldContext = field.getAnnotation(FieldContext.class);
-            if (fieldContext == null) {
+            if (fieldContext == null || fieldContext.deprecated()) {
```
Comment (horizonzy) [inline pulsar-broker-common/src/main/java/org/apache/pulsar/broker/BaseGenerateDocumentation.java:93]: @urfreespace Pls help check this change, shall we ignore the FileContext.deprecated param.

Comment (urfreespace) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/BaseGenerateDocumentation.java:93]: @Anonymitaet any idea? I think it's not a tech question

**Pull request merged**